simulator/risc_v/t/unit_test.cpp

```diff
@@ -275,3 +275,22 @@ TEST_CASE( "RISV RV64 pack")
     instr.execute();
     CHECK( instr.get_v_dst() == 0x3333'3333'2222'2222);
 }
+
+TEST_CASE("RISCV RV32 rol")
+{
+    CHECK( RISCVInstr<uint32>(0x411815b3).get_disasm() == "rol $a1, $a6, $a7");
+    RISCVInstr<uint32> instr( "rol", 0);
+    instr.set_v_src( 0x17013f, 0);
+    instr.set_v_src( 0xf2, 1);
+    instr.execute();
+    CHECK( instr.get_v_dst() == 0x4fc005c);
+}
+
+TEST_CASE("RISCV RV64 rol")
+{
+    RISCVInstr<uint64> instr( "rol", 0);
+    instr.set_v_src( 0x17013f, 0);
+    instr.set_v_src( 0xf2, 1);
+    instr.execute();
+    CHECK( instr.get_v_dst() == 0x4fc005c);
+}
```

Comment:
`0xf2` looks quite complicated, can we start with simple scenarios like 1 or 4?
Could you please also check corner cases: rotating by 0, rotating by XLEN?

Reply:
Did my best, hope it's better now.

Reply:
Thanks. Yes, that's very well!